tests/test_operation.py

```diff
@@ -92,6 +92,24 @@ class DummyOp(qml.operation.Operator):
         op.name = "MyOp"
         assert op.name == "MyOp"
 
+    def test_default_hyperparams(self):
+        """Tests that the hyperparams attribute is defined for all operations."""
```

Comment:
Might also be good to add a test that _adds_ hyperparameters, and checks that it works correctly?

Although, how do we want the UI to look?

- A setter, `self.hyperparameters = {...}`?
- Something you pass to the base class on init? `super().__init__(params, wires, hyperparams)`?

Reply:
I thought that the init method just saves all relevant inputs in the dictionary, like below...

Reply:
Ah I see! It might make sense then to set a setter method for `self.hyperparameters`, if only so that we can document it (it feels a bit odd to have developers set a private attribute).

Reply:
Given the mutability of dictionaries, doing something like
```
op.hyperparameters['key'] = value
```
will update the private dictionary `op._hyperparameters`.  

So we don't need a setter to be able to modify the dictionary at any point.  We can't reset the dictionary or turn it into a different type of variable without using the private `_hyperparameters`, but we can still update it.